### ISSUE TITLE
No AVIF support in v1.6.1+

### ISSUE THREAD

**Issue by gaxweb** [flathub/com.xnview.XnViewMP#50]
It's supposed to be supported since v1.6.1: https://newsgroup.xnview.com/viewtopic.php?t=46266

But even in v1.6.2 I can't view or save them. I'm on Fedora 39.

**Comment by gaxweb** [comment]
It's an issue with the non-flatpak version, too, I've found.

**Comment by gaxweb** [comment]
Ok, so it is a packaging problem with the original software which creates this problem. An strace with a grep on libheif reveals the following for the TAR version:

```
openat(AT_FDCWD, "/home/gaxweb/Downloads/XnViewMP-linux-x64/XnView/lib/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/home/gaxweb/Downloads/XnViewMP-linux-x64/XnView/Plugins/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "glibc-hwcaps/x86-64-v3/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "glibc-hwcaps/x86-64-v2/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/lib64/glibc-hwcaps/x86-64-v3/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/lib64/glibc-hwcaps/x86-64-v2/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/lib64/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/usr/lib64/glibc-hwcaps/x86-64-v3/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/usr/lib64/glibc-hwcaps/x86-64-v2/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/usr/lib64/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/home/gaxweb/Downloads/XnViewMP-linux-x64/XnView/lib/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/home/gaxweb/Downloads/XnViewMP-linux-x64/XnView/Plugins/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "glibc-hwcaps/x86-64-v3/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "glibc-hwcaps/x86-64-v2/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/lib64/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
openat(AT_FDCWD, "/usr/lib64/libheif.so", O_RDONLY|O_CLOEXEC) = -1 ENOENT (File or folder not found)
```

Problem appears to be the filename. If I copy my "libheif.so.1.17.5" to "XnView/lib/libheif.so", it works. The "libheif.so" file without the .1 at the end is actually only contained in the `-dev` package on Fedora. Normal (non-dev) users won't have that installed.

Anyway, it doesn't work with the Flatpak version for me either.

**Comment by gombosg** [comment]
Correct! In Fedora, all libraries have soversions in their file name. The "soversion-less" file name is always the latest development version. When you create an RPM file and compile the app, it always links as a dependency against the correct [soname](https://docs.fedoraproject.org/en-US/packaging-guidelines/#_soname_handling).

Not sure how to change this, as XNView is closed source, they should link against versioned library files. It has been [reported](https://newsgroup.xnview.com/viewtopic.php?f=62&t=46480&p=196828&hilit=libheif#p196828) so we could possibly wait for a fix?

I'll look into a packaging fix as we already do [something similar in the manifest](https://github.com/flathub/com.xnview.XnViewMP/blob/5fa9dabafe3aaab485ac0db0cb75195871df4988/XnViewMP.yaml#L17).

**Comment by DanMan** [comment]
Well, they didn't move the thread to the TODO forum and the thread is months old by now and there have been releases inbetween that didn't fix it, so… I don't expect them to do something about it, because it seems to work on Ubuntu still.

And yes, I saw the manifest, too, so I thought it might be a quick fix on your side.

**Comment by gaxweb** [comment]
Well, they didn't move the months old thread to the TODO forum and there have been releases in between that didn't fix it, so… I don't expect them to do something about it, because it seems to work on Ubuntu still.

And yes, I saw the manifest, too, that's why I thought it might be a quick fix on your side.

**Comment by gombosg** [comment]
Turns out that Flatpak has no access to system libraries. Well, that's its goal after all, to provide a sandboxed and unified environment.
I checked, for example GIMP has HEIF support obviously. Apparently, it builds [libheif](https://github.com/flathub/org.gimp.GIMP/blob/1cb7830d38ca209b5f3b7531b08a497ebdddf803/org.gimp.GIMP.json#L727-L871) as part of creating the flatpak.

I'll try to see based on [similar examples](https://github.com/search?q=org%3Aflathub%20libheif&type=code), how hard it is to do this with XNView as well.

**Comment by gombosg** [comment]
@gaxweb can you test this? ☝🏽 
For me it opens AVIF, but not HEIC files, even though `libheif.so` is now in the `build/lib` folder. I wonder what else may be missing. 🤔 

LOL BTW `libheif` is just [being added](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/commits/master?search=libheif) to the freedesktop SDK, it might make it to the next release.

**Comment by DanMan** [comment]
> LOL BTW libheif is just [being added](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/commits/master?search=libheif) to the freedesktop SDK, it might make it to the next release

Nice, but that wouldn't help with this, right? Because of the file name?

Build works for me, too, with AVIF files. I vaguely remember some other issue regarding HEIC/HEIF images, but I can't find it anymore. AVIF's good enough for me anyway.

Cheers!

**Comment by gaxweb** [comment]
> LOL BTW libheif is just [being added](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/commits/master?search=libheif) to the freedesktop SDK, it might make it to the next release

Nice, but that wouldn't help with this, right? Because of the file name?

Build works for me, too, with AVIF files. I vaguely remember some other issue regarding HEIC/HEIF images, but I can't find it anymore. AVIF's good enough for me anyway.

Cheers!

**Comment by gombosg** [comment]
OK then, thanks, merged for a release.
